docker/web-and-data/Dockerfile

```diff
@@ -0,0 +1,39 @@
+# This image includes both the web app and the data import related scripts in
+# one image. Because of historic reasons some of these scripts run java
+# artifacts under the hood. It therefore currently makes more sense to have a
+# single image supporting both instead of moving the scripts out to a separate
+# container.
+FROM maven:3.5.4 as build
+COPY $PWD /cbioportal
```

Comment:
Do the docs for multi-stage build explicitly state that the potentially very large amount of private branch data and configuration in the `.git/` folder will _not_ be included in the final Docker image?

Reply:
good point - that's prolly why maven could correctly determine the git tags 😄

Reply:
@fedde-s could you elaborate on this one actually -  We're copying over the `war` and `webapp-runner` from the first image specifically, so the final image will only contain that.

Im not sure what the best procedure is here. We pull a bunch of git info using mvn for https://www.cbioportal.org/api/info, so it's useful to have `.git` available during the compile step. We could potentially pass these as env variables as well

Reply:
My worry is that `docker history --no-trunc <image name>`, `docker inspect <image name>` or `docker inspect <container name>` or similar metadata digs into the image might leak information about the contents of the `.git/` folder other than the version description that the build process needs to include in the .war file.

If we can find some docs or practical evidence implying that the information really isn't preserved in the final image, I agree that giving the ‘mvn stage’ of the Dockerfile access is much better than passing `--build-args`, so that a conventional `docker build` command just works. Otherwise, a few lines of shell script or a documented instruction to supply the build arg could imaginably be a big security improvement for some use-cases.

Reply:
@fedde-s Thanks for clarifying! That is a good point

I couldn't find an answer using google so opened a StackOverflow question: https://stackoverflow.com/questions/56278325/is-it-a-security-risk-to-include-a-git-folder-in-the-build-step-of-a-docker-mul. Please help me rephrase it if you think it's unclear

The automated image builds would be done on CircleCI or Docker Hub, so I don't think there's a risk there, since it will only pull public git info. But maybe if you build it locally and push it publicly it could result in some .git folder leaking. Hopefully the StackOverflow question can give an answer to that.

I could add some warning as comments to the image referring to that question for now. Is that an ok solution?

Reply:
Yes! That would definitely do, and resolve my objection to merging this :+1:

Reply:
updated